File: python/phase 1-condtional_thinking/q12_triangle_type.py
"""
Phase 1: Conditional Thinking (If-Else, Boolean Logic)
Level 2: Nested If & Multiple Conditions
Question 12: If the sides form a valid triangle, determine whether it is equilateral, isosceles, or scalene.

Mentor Tips (Python 3.14+):
- First check if it is a valid triangle (Q11). If so: 3 equal sides = equilateral, 2 equal sides = isosceles, 0 equal sides = scalene.
"""

def solve(s1: float, s2: float, s3: float) -> str:
    # TODO: Implement this method to solve the question
    if s1+s2>s3 and s2+s3>s1 and s1+s3>s2:
        if s1 == s2 == s3:
            return "equilateral"
        elif s1 == s2 or s2 == s3 or s1 == s3:
            return "isosceles"
        else:
            return "scalene"
    else:
        return "not a valid triangle"

def main():
    s1 = float(input("Side 1: "))
    s2 = float(input("Side 2: "))
    s3 = float(input("Side 3: "))
    print(f"Triangle type: {solve(s1, s2, s3)}")

if __name__ == "__main__":
    main()

File: java/src/phase 1-conditional_thinking_java/Q12TriangleType.java
import java.util.Scanner;

/**
 * Phase 1: Conditional Thinking (If-Else, Boolean Logic)
 * Level 2: Nested If & Multiple Conditions
 * Question 12: If the sides form a valid triangle, determine whether it is equilateral, isosceles, or scalene.
 * 
 * Mentor Tips (Java 26):
 * - First check if it is a valid triangle (Q11). If so: 3 equal sides = equilateral, 2 equal sides = isosceles, 0 equal sides = scalene.
 */
public class Q12TriangleType {

    public static String solve(double s1, double s2, double s3) {
        // TODO: Implement this method to solve the question
        return "invalid";
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Side 1: "); double s1 = scanner.nextDouble();\n        System.out.print("Side 2: "); double s2 = scanner.nextDouble();\n        System.out.print("Side 3: "); double s3 = scanner.nextDouble();\n        System.out.println("Triangle type: " + solve(s1, s2, s3));
        scanner.close();
    }
}
